File: Data Structures & Algorithms/max-water-container/submission-0.js
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(heights) {
        let currentMax = 0;
	    let left = 0;
	    let right = heights.length - 1;
	    while (left < right) {
		    currentMax = Math.max(currentMax, ((right - left) * Math.min(heights[left], heights[right])));
		    if (heights[left] >= heights[right]) {
			    right--;
		    } else {
			    left++;
		    }
	    }
	    return currentMax;	
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        left = 0
        right = len(nums) - 1

        if nums[left] <= nums[right]:
            return nums[left]

        while left < right:
            mid = left + (right - left) // 2

            if nums[right] < nums[mid]:
                left = mid + 1

            else:
                right = mid
        
        return nums[left]
